### PR TITLE
CI/CD: Refactor building docker images

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
- Fixes the current failures in nightly docker image builds.
- Unifies the following docker-related workflows into one workflow file:
  - docker build `.github/workflows/ci-pytorch_dockers.yml`
  - docker build & push `.github/workflows/events-nightly.yml`

### Motivation
- This PR reduces the risk of pushing docker images with the wrong build arguments. For example, in master, there's a discrepancy between our CI and CD:
https://github.com/Lightning-AI/lightning/blob/61c28cb428a13c2aea6d7f3f55e0f00431a4ea4e/.github/workflows/ci-pytorch_dockers.yml#L58-L59
https://github.com/Lightning-AI/lightning/blob/61c28cb428a13c2aea6d7f3f55e0f00431a4ea4e/.github/workflows/events-nightly.yml#L70-L71

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [rfc] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-pytorch_dockers.yml [hunk 1]
```diff
@@ -86,98 +94,149 @@ jobs:
...
+        with:
+          status: ${{ job.status }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          notification_title: ${{ format('CUDA; {0} py{1} for *{2}*', runner.os, matrix.python_version, matrix.pytorch_version) }}
+          message_format: '{emoji} *{workflow}* {status_message}, see <{run_url}|detail>, cc: <@U01A5T7EY9M>'  # akihironitta
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
 
-  build-Conda:
+  conda:
```
Comment (Borda) [inline .github/workflows/ci-pytorch_dockers.yml:125]: I would keep the verb so it won't collide with test which uses also conda...

**Review by Borda — COMMENTED** [review]
let's rename it as `ci-cd-pytorch_dockers.yml` as it covers also releasing

**Review by carmocca — APPROVED** [review]
You might want to consider fixing the commit history and doing a rebase and merge to maintain the file's git history. Or renaming the file in a separate PR

LGTM either way!

**Review by rohitgr7 — APPROVED** [review]
I'd say rebase and merge... if not too hard.

**Review by Borda — APPROVED** [review]

**Pull request merged**